simphony/cuds/primitive_cell.py

```diff
@@ -0,0 +1,537 @@
+import numpy as np
+from enum import Enum
+
+
+class BravaisLattice(Enum):
+    """The 3D Bravais lattices"""
+    CUBIC = 0
+    BODY_CENTERED_CUBIC = 1
+    FACE_CENTERED_CUBIC = 2
+    RHOMBOHEDRAL = 3
+    HEXAGONAL = 4
+    TETRAGONAL = 5
+    BODY_CENTERED_TETRAGONAL = 6
+    ORTHORHOMBIC = 7
+    BODY_CENTERED_ORTHORHOMBIC = 8
+    FACE_CENTERED_ORTHORHOMBIC = 9
+    BASE_CENTERED_ORTHORHOMBIC = 10
+    MONOCLINIC = 11
+    BASE_CENTERED_MONOCLINIC = 12
+    TRICLINIC = 13
+
+
+class PrimitiveCell(object):
+    """
+    A primitive cell of a Bravais lattice.
+    The cell edges are represented with primitive vectors.
+    
+    Primitive cells can be configured either by providing the
+    primitive vectors explicitly (default) or by providing appropriate
+    (conventional) unit cell edge parameters (for the class methods).
+
+    Attributes
+    ----------
+    p1, p2, p3: 3 x 3D vector
+        primitive vectors
+
+    bravais_lattice: BravaisLattice(Enum)
+        the 3D Bravais lattice for which the primitive cell is defined,
+        i.e. not the symmetry group or type of the primitive cell itself
+    """
+   
+    def __init__(self, p1, p2, p3):
+        """
+        Configure an arbitrary primitive cell.
+        The cell edges are represented with primitive vectors.
+        
+        Parameters
+        ----------
+        p1, p2, p3: 3 x 3D vector
+            primitive vectors
+
+        Raises
+        ------
+        ValueError
+           if an edge length is not strictly positive,
+           if the edges are parallel to each other,
+           if the volume of the cell is zero
+        """
+        a = np.sqrt(np.dot(p1, p1))
+        b = np.sqrt(np.dot(p2, p2))
+        c = np.sqrt(np.dot(p3, p3))
+        
+        if np.any(np.isclose((a, b, c), (0, 0, 0))):
+            message = 'Edge lengths must be strictly positive'
+            raise ValueError(message)
+
+        cos1 = np.dot(p2, p3)/(b*c)
+        cos2 = np.dot(p1, p3)/(a*c)
+        cos3 = np.dot(p1, p2)/(a*b)
+
+        if np.any(np.isclose(np.absolute((cos1, cos2, cos3)), (1, 1, 1)):
+            message = 'Edges must not be parallel to each other'
+            raise ValueError(message)
+        
+        self._p1 = p1
+        self._p2 = p2
+        self._p3 = p3
+
+        if np.isclose(self.volume(), 0):
+            message = 'Cell volume must be non-zero'
+            raise ValueError(message)
+
+        self._bravais_lattice = BravaisLattice.TRICLINIC
+            
+    @classmethod
+    def create_cell_cubic_lattice(cls, a):
```

Comment:
Looking at the methods I would suggest to rename the `classmethods` just with the name of the cell e.g:
```
def cubic(cls, a)
def body_centered_cubic(cls)   # we should avoid cutting the words short
```
This will make the functions short and easier to understand

```
cell = PrimitiveCell.cubic(10)
```